packages/wallet/src/util.ts

```diff
@@ -1 +1,158 @@
+
+        if (!address || !parsedAddress) {
+          logger.error('could not get change address');
+          throw new ApiError(500, 'could not get change address');
+        } else {
+          return Buffer.from(parsedAddress.to_bytes()).toString('hex');
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getNetworkId: async (): Promise<number> => {
+      logger.debug('getting networkId');
+      return Promise.resolve(props.networkId);
+    },
+    getRewardAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting reward addresses');
+      try {
+        const [{ rewardAccount }] = await firstValueFrom(wallet.addresses$);
+        const parsedAddress = parseCslAddress(rewardAccount as unknown as string);
+
+        if (!rewardAccount || !parsedAddress) {
+          throw new ApiError(500, 'could not get reward address');
+        } else {
+          return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getUnusedAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting unused addresses');
+      return Promise.resolve([]);
+    },
+    getUsedAddresses: async (_paginate?: Paginate): Promise<Cbor[]> => {
+      logger.debug('getting changeAddress');
+
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address as unknown as string);
+
+      if (!address || !parsedAddress) {
+        throw new ApiError(500, 'could not get used addresses');
+      } else {
+        return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+      }
+    },
+    getUtxos: async (amount?: Cbor, paginate?: Paginate): Promise<Cardano.Utxo[] | undefined> => {
+      const utxos = await firstValueFrom(wallet.utxo.available$);
+
+      // eslint-disable-next-line unicorn/no-useless-undefined
+      if (!utxos) return Promise.resolve(undefined);
+      const cslUtxos = coreToCsl.utxo(utxos);
+
+      if (amount) {
+        try {
+          const filterAmount = CSL.Value.from_bytes(Buffer.from(amount, 'hex'));
+          const filteredUtxosByAmount = cslUtxos.filter(
+            (unspent) =>
+              !unspent.output().amount().compare(filterAmount) || unspent.output().amount().compare(filterAmount) !== -1
+          );
+          logger.debug(filteredUtxosByAmount);
+        } catch (error) {
+          logger.debug(error);
+          throw new ApiError(400, error);
+        }
+      }
+
+      if (paginate) {
+        return Promise.resolve(
+          cslUtxos.slice(
+            paginate.page * paginate.limit,
+            paginate.page * paginate.limit + paginate.limit
+          ) as unknown as Cardano.Utxo[]
+        );
+      }
+
+      return Promise.resolve(cslUtxos as unknown as Cardano.Utxo[]);
```

Comment:
This seems off. the return type should probably be `string[]`? `cslUtxos.map(cslUtxo => Buffer.from(cslUtxo.to_bytes()).toString('hex')`